torch/_dynamo/variables/torch.py

```diff
@@ -712,6 +712,16 @@ def register_as_subgraph(fn, name, args):
             # ops - see torch/dispatch/_dispatcher.py
             from .. import config
 
+            # The current recursive export() implementation will
+            # not "see" any side effect updates from the enclosing
+            # context, which can result in possibly incorrect
+            # export.  This assert ensures that there were no
+            # outstanding side effects at the time cond() was called.
+            #
+            # TODO: This assert may be too aggressive; I'm landing it
+            # to see if it is or not.
+            assert tx.output.side_effects.is_empty()
```

Comment:
I think this is good, would you be able to add a message to the rhs of the assert?

Reply:
Alternatively, you can throw this at the top of export, if you like.